spock-specs/src/test/groovy/org/spockframework/mock/runtime/ByteBuddyMockMakerSpec.groovy

```diff
@@ -77,4 +77,72 @@ class ByteBuddyMockMakerSpec extends Specification {
     def ex = thrown(CannotCreateMockException)
     ex.message == "Cannot create mock for class java.lang.StringBuilder. byte-buddy: Cannot mock final classes."
   }
+
+  def "Mock class"() {
+    when:
+    DataClass s = Mock(mockMaker: MockMakers.byteBuddy)
+    then:
+    !s.boolField
+    s.stringField == null
```

Comment:
❔ Isn't it a bit confusing calling it field when we are using the getters?
It's also conflating the tests a bit with property handling. I think we should use `bool()` and `string()` instead.